File: c-sharp-design-patterns/OOP Principles/Inheritance/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace c_sharp_design_patterns.OOP_Principles.Inheritance
{
    public class Car : Vehicle
    {
        public int NumberOfDoors { get; }

        public Car(string brand, string model, int year, int numberOfDoors) : base(brand, model, year)
        {
            NumberOfDoors = numberOfDoors;
        }
    }
}

File: OOP_Principles/Inheritance/Car.java
package OOP_Principles.Inheritance;

public class Car extends Vehicle {

  private int numberOfDoors;

  public Car(String brand, String model, int year, int numberOfDoors) {
    super(brand, model, year);
    this.numberOfDoors = numberOfDoors;
  }

  @Override
  public String toString() {
    return String.format(
      "%d %s %s with %d doors",
      super.year,
      super.brand,
      super.model,
      this.numberOfDoors
    );
  }
}
